lib/modules/dosomething/dosomething_reportback/includes/ReportbackItemTransformer.php

```diff
@@ -40,11 +40,26 @@ public function index($parameters) {
   public function show($id) {
     try {
       $reportbackItem = ReportbackItem::get($id);
-      $reportbackItem = $this->removeUnauthorizedResults($reportbackItem);
-      $reportbackItem = services_resource_build_index_list($reportbackItem, 'reportback-items', 'id');
+
+      if (!user_access('administer modules')) {
+        $reportbackItem = $this->removeUnauthorizedResults($reportbackItem);
+      }
+
+      if ($reportbackItem) {
+        $reportbackItem = services_resource_build_index_list($reportbackItem, 'reportback-items', 'id');
+      } else {
+        throw new Exception('Access denied.');
```

Comment:
Instead of throwing the `Access denied` exception here, we could also do that in the `removeUnauthorizedResults()`. If there are no results passed to `removeUnauthorizedResults()` as the `$data`, then that means no items found so then immediately throw the `Exception('No reportback items data found.'). If there are items, but they get removed due to access restrictions, then change `Exception('No reportback items data found.')` to `Exception('Access denied.')`.